spec/services/schedule_data_manager_service_spec.rb

```diff
@@ -21,13 +21,125 @@
   describe '#holidays' do
     let!(:school)                                    { create(:school, calendar: calendar) }
     let(:date_version_of_holiday_date_from_calendar) { Date.parse(random_before_holiday_start_date) }
+    let!(:service)          { ScheduleDataManagerService.new(school) }
 
     it 'assigns school date periods for the analytics code' do
+      allow(school).to receive(:minimum_reading_date).and_return(nil)
       results = ScheduleDataManagerService.new(school).holidays
       school_date_period = results.find_holiday(date_version_of_holiday_date_from_calendar)
       expect(school_date_period.start_date).to eq date_version_of_holiday_date_from_calendar
       expect(school_date_period.type).to_not be_nil
     end
+
+    it 'loads holiday data' do
```

Comment:
Can you expand the tests to check the return type of the objects. Part of the contract with the analytics is that we provide instances of its object, rather than just hashes.